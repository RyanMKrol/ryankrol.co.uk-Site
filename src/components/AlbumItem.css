.AlbumItem {
  padding: 15px;
  max-width: 300px;
}

.album-item-thumbnail {

}

.album-item-image {
  max-width: 95%;
}

.album-item-title {
  max-width: 95%;
}
