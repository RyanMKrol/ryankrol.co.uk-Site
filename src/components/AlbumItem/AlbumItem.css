.AlbumItem {
  padding: 15px;
  max-width: 300px;
  text-align: left;
  background-color: #eee;
  margin: 10px;
  padding-bottom: 0;
  font-family: 'Archivo Narrow', sans-serif;
  font-size: 14px;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
}

.album-item-thumbnail {
  text-align: center;
  align-items: center;
}

.album-item-image {
  max-width: 100%;
  display: flex;
  margin: auto;
  border-radius: 5px;
}

.album-information {
  display: flex;
  flex: 1;
}

.album-information-content {
  margin: auto 0;
  padding: 16px 0;
}

.album-information-key {
  color: #555;
}

.album-information-value {
  color: #888;
}
