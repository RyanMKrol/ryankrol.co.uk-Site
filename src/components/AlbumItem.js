import React, { Component } from 'react'

import './AlbumItem.css';

class AlbumItem extends Component {
  constructor(props) {
    super()

    this.artist = props.artist
    this.albumName = props.albumName
    this.thumbnail = props.thumbnail
    this.albumLink = props.albumLink

    if (!(this.artist && this.albumName && this.thumbnail && this.albumLink)) {
      throw new Error('Did not pass all required args for rendering AlbumItem')
    }
  }

  render() {
    return (
      <div className="AlbumItem">
        <div className="album-item-thumbnail">
          <a href={this.albumLink}>
            <img
              className="album-item-image"
              src={this.thumbnail}
              alt={`${this.artist} - ${this.albumName}`}
            />
          </a>
        </div>
        <div className="album-item-title">
          <a href={this.albumLink}>
            {this.artist} - {this.albumName}
          </a>
        </div>
      </div>
    )
  }
}

export default AlbumItem
