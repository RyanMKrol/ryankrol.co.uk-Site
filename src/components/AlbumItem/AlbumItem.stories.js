import React from 'react';
import AlbumItem from './AlbumItem';

export default {
  title: 'AlbumItem',
};

/**
 * default story
 *
 * @returns {string} JSX for the component I want to display
 */
export const normal = () => (
  <div style={{ display: 'flex' }}>
    <AlbumItem
      artist={'Die Antwoord'}
      albumName={'Ten$ion'}
      thumbnail={
        'https://lastfm.freetls.fastly.net/i/u/300x300/d3561c0bd8c01a27bc3e8991a68f45f4.png'
      }
      albumLink={'/'}
      playcount={123}
    />
  </div>
);
