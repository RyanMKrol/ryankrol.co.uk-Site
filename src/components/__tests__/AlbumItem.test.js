import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AlbumItem from './../AlbumItem'

const defaultArgs = {
  artist: 'test-artist',
  albumName: 'test-albumName',
  thumbnail: 'test-thumbnail',
  albumLink: 'test-albumLink',
  playcount: 'test-playcount',
}

// Render Tests

it('renders AlbumItem correctly', () => {
    const albumItem = shallow(
      <AlbumItem {...defaultArgs} />)
    expect(toJson(albumItem)).toMatchSnapshot()
})

it('throws when required parameters are omitted', () => {
    expect(() => mount(<AlbumItem />)).toThrow()
})

it('does not throw when required parameters are provided', () => {
    expect(() => mount(<AlbumItem {...defaultArgs}/>)).not.toThrow()
})
