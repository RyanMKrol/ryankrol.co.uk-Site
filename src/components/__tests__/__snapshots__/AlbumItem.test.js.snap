// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AlbumItem correctly 1`] = `
<div
  className="AlbumItem"
>
  <div
    className="album-item-thumbnail"
  >
    <a
      href="test-albumLink"
    >
      <img
        className="album-item-image"
        src="test-thumbnail"
      />
    </a>
  </div>
  <div
    className="album-item-title"
  >
    <a
      href="test-albumLink"
    >
      test-artist
       - 
      test-albumName
    </a>
  </div>
</div>
`;
